File: app/src/main/java/com/yijian/staff/mvp/forgetpassword/ForgetPasswordActivity.java
package com.yijian.staff.mvp.forgetpassword;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.yijian.staff.R;
import com.yijian.staff.net.httpmanager.HttpManager;
import com.yijian.staff.net.response.ResultJSONObjectObserver;
import com.yijian.staff.util.CommonUtil;
import com.yijian.staff.widget.NavigationBar2;

import org.json.JSONObject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class ForgetPasswordActivity extends AppCompatActivity {

    @BindView(R.id.et_account)
    EditText etAccount;
    @BindView(R.id.et_phonenum)
    EditText etPhonenum;

    @BindView(R.id.et_code)
    EditText etCode;
    @BindView(R.id.tv_getcode)
    TextView tvGetcode;
    @BindView(R.id.et_passwd)
    EditText etPasswd;
    @BindView(R.id.et_re_passwd)
    EditText etRePasswd;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_forget_password);
        ButterKnife.bind(this);

        NavigationBar2 navigationBar2 = (NavigationBar2) findViewById(R.id.forget_password_activity_navigation_bar2);
        navigationBar2.setTitle("忘记密码");
        navigationBar2.hideLeftSecondIv();
        navigationBar2.setBackClickListener(this);

    }

    @OnClick({R.id.tv_getcode, R.id.btn_send})
    public void onViewClicked(View view) {

        String account = etAccount.getText().toString();
        String telephone = etPhonenum.getText().toString().trim();
        String verificationCode = etCode.getText().toString().trim();
        String newPwd = etPasswd.getText().toString().trim();
        String confirmPwd = etRePasswd.getText().toString().trim();
        switch (view.getId()) {
            case R.id.tv_getcode:

                if (TextUtils.isEmpty(account) || TextUtils.isEmpty(telephone)) {
                    Toast.makeText(ForgetPasswordActivity.this, "账号和手机号不能为空", Toast.LENGTH_SHORT).show();
                    return;
                } else {
                    if (CommonUtil.isPhoneFormat(telephone)) {
                        HttpManager.getCode(account, telephone, new ResultJSONObjectObserver() {
                            @Override
                            public void onSuccess(JSONObject result) {


                            }

                            @Override
                            public void onFail(String msg) {

                            }
                        });
                    } else {
                        Toast.makeText(ForgetPasswordActivity.this, "输入的手机号不正确,请重新输入！", Toast.LENGTH_SHORT).show();
                    }
                }

                break;
            case R.id.btn_send:
                if (TextUtils.isEmpty(account) ) {
                    Toast.makeText(ForgetPasswordActivity.this, "账号不能为空", Toast.LENGTH_SHORT).show();
                    return;
                } else if ( TextUtils.isEmpty(telephone)) {
                    Toast.makeText(ForgetPasswordActivity.this, "手机号不能为空", Toast.LENGTH_SHORT).show();
                    return;
                } else if (TextUtils.isEmpty(verificationCode)) {
                    Toast.makeText(ForgetPasswordActivity.this, "验证码不能为空", Toast.LENGTH_SHORT).show();
                    return;
                } else if (TextUtils.isEmpty(newPwd)||TextUtils.isEmpty(confirmPwd) ) {
                    Toast.makeText(ForgetPasswordActivity.this, "密码不能为空", Toast.LENGTH_SHORT).show();
                    return;
                } else {
                    if (CommonUtil.isPhoneFormat(telephone)) {
                        HttpManager.resetPassword(account, telephone, verificationCode,newPwd,confirmPwd,new ResultJSONObjectObserver() {
                            @Override
                            public void onSuccess(JSONObject result) {


                            }

                            @Override
                            public void onFail(String msg) {
                                Toast.makeText(ForgetPasswordActivity.this, msg, Toast.LENGTH_SHORT).show();

                            }
                        });
                    } else {
                        Toast.makeText(ForgetPasswordActivity.this, "输入的手机号不正确,请重新输入！", Toast.LENGTH_SHORT).show();
                    }
                }
                break;
        }
    }
}

File: app/src/main/java/com/yijian/staff/mvp/forgetpassword/ForgetPasswordActivity.kt
package com.yijian.staff.mvp.forgetpassword

import android.os.Bundle
import android.text.TextUtils
import android.view.View


import com.yijian.staff.R
import com.yijian.staff.mvp.base.mvc.MvcBaseActivity
import com.yijian.staff.net.httpmanager.HttpManager
import com.yijian.staff.net.response.ResultJSONObjectObserver
import com.yijian.staff.util.CommonUtil
import com.yijian.staff.util.CountDownTimerUtils
import com.yijian.staff.widget.NavigationBar

import org.json.JSONObject

import butterknife.OnClick
import kotlinx.android.synthetic.main.activity_forget_password.*

class ForgetPasswordActivity : MvcBaseActivity() {


    override fun getLayoutID(): Int {
        return R.layout.activity_forget_password
    }

    override fun initView(savedInstanceState: Bundle?) {
        val navigationBar = findViewById<View>(R.id.forget_password_activity_navigation_bar2) as NavigationBar
        navigationBar.setTitle("忘记密码")
        navigationBar.hideLeftSecondIv()
        navigationBar.setBackClickListener(this)
    }

    @OnClick(R.id.tv_getcode, R.id.btn_send)
    fun onViewClicked(view: View) {

        val account = et_account.text.toString()
        val telephone = et_phonenum.text.toString().trim { it <= ' ' }
        val verificationCode = et_code.text.toString().trim { it <= ' ' }
        val newPwd = et_passwd.text.toString().trim { it <= ' ' }
        val confirmPwd = et_re_passwd.text.toString().trim { it <= ' ' }
        when (view.id) {
            R.id.tv_getcode ->

                if (TextUtils.isEmpty(account) || TextUtils.isEmpty(telephone)) {
                    showToast("账号和手机号不能为空!")
                    return
                } else {
                    if (CommonUtil.isPhoneFormat(telephone)) {
                        tv_getcode.isEnabled = false
                        val countDownTimerUtils = CountDownTimerUtils(tv_getcode, 30000, 1000)
                        countDownTimerUtils.start()
                        showLoading()
                        HttpManager.getCode(account, telephone, object : ResultJSONObjectObserver(lifecycle) {
                            override fun onSuccess(result: JSONObject) {
                                showToast("验证码已发送!")
                                hideLoading()
                            }

                            override fun onFail(msg: String) {
                                tv_getcode.isEnabled = true
                                countDownTimerUtils.cancel()
                                countDownTimerUtils.onFinish()
                                showToast(msg)
                                hideLoading()

                            }
                        })
                    } else {
                        showToast("输入的手机号不正确,请重新输入!")

                    }
                }
            R.id.btn_send -> {
                if (TextUtils.isEmpty(account)) {
                    showToast("账号不能为空!")
                    return
                }
                if (TextUtils.isEmpty(telephone)) {
                    showToast("手机号不能为空!")
                    return
                }
                if (TextUtils.isEmpty(verificationCode)) {
                    showToast("验证码不能为空!")
                    return
                }
                if (TextUtils.isEmpty(newPwd) || TextUtils.isEmpty(confirmPwd)) {
                    showToast("密码不能为空!")
                    return
                }
                if (newPwd != confirmPwd) {
                    showToast("2次输入的密码不同!")
                    return
                }
                if (!CommonUtil.isPassWordFormat(newPwd)) {
                    showToast("新密码格式不正确,密码是数字和字母的6-20位组合！")
                    return
                }
                if (!CommonUtil.isPassWordFormat(confirmPwd)) {
                    showToast("确认密码格式不正确，密码是数字和字母的6-20位组合！")
                    return
                }
                if (CommonUtil.isPhoneFormat(telephone)) {
                    showLoading()
                    HttpManager.resetPassword(account, telephone, verificationCode, newPwd, confirmPwd, object : ResultJSONObjectObserver(lifecycle) {
                        override fun onSuccess(result: JSONObject) {
                            hideLoading()
                            finish()
                        }

                        override fun onFail(msg: String) {
                            showToast(msg)
                            hideLoading()
                        }
                    })
                } else {
                    showToast("输入的手机号不正确,请重新输入！")
                }
            }
        }
    }
}
